repository/audit.go

```diff
@@ -98,21 +136,36 @@ func (r *AuditRepository) CreateAuditLogAccountLevel(
 		auditLog.OwnerID = *ownerId
 	}
 
-	res := r.DB.Create(&auditLog)
+	for _, detail := range details {
+		auditLog.Details = append(auditLog.Details, AuditLogDetail{
+			ID:      uuid.New(),
+			Message: detail,
+		})
+	}
+
+	r.PushToAuditQueue(auditLog)
+	return auditLog, nil
+}
 
-	return auditLog, res.Error
+func (r *AuditRepository) PushToAuditQueue(auditLog AuditLog) {
+	r.queueLock.Lock()
+	r.auditLogQueue = append(r.auditLogQueue, auditLog)
+	r.queueLock.Unlock()
 }
 
-func (r *AuditRepository) CreateAuditLogDetail(
-	parentId uuid.UUID,
-	message string,
-) (AuditLogDetail, error) {
-	auditLogDetail := AuditLogDetail{
-		ID:         uuid.New(),
-		AuditLogID: parentId,
-		Message:    message,
+func (r *AuditRepository) FlushAuditLogQueue() error {
+	// Create a deep copy of the previous queue.
+	r.queueLock.Lock()
+	var oldQueue []AuditLog = make([]AuditLog, 0)
+	for _, log := range r.auditLogQueue {
+		var copyLog AuditLog = log
+		copy(copyLog.Details, log.Details)
+		oldQueue = append(oldQueue, copyLog)
 	}
+	r.auditLogQueue = make([]AuditLog, 0)
+	r.queueLock.Unlock()
 
-	err := r.DB.Create(&auditLogDetail).Error
-	return auditLogDetail, err
+	batchSize := 50
```

Comment:
Same comment about this hardcoded - create a global constant default value for it